webapp/src/views/projects/translations/Screenshots/Screenshots.tsx

```diff
@@ -0,0 +1,288 @@
+  & .arrowRight:hover {
+    opacity: 0.8;
+    color: ${({ theme }) => theme.palette.tokens.icon.onDarkHover};
+    background-color: ${({ theme }) =>
+      theme.palette.tokens.icon.backgroundDarkHover};
+  }
+`;
+
+const StyledScrollWrapper = styled(Box)`
+  padding: 0px 6px;
+  padding-top: 12px;
+  overflow-y: auto;
+  display: grid;
+  grid-auto-flow: column;
+  justify-content: start;
+  gap: 4px;
+  scrollbar-width: none;
+  scroll-snap-type: x mandatory;
+`;
+
+type Props = {
+  screenshots: ScreenshotModel[];
+  keyId: number;
+  oneBig?: boolean;
+  width?: number;
+  sx?: SxProps;
+};
+
+export const Screenshots = ({
+  screenshots,
+  keyId,
+  oneBig,
+  width,
+  sx,
+}: Props) => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const size = width && Math.floor(width / 50) * 50;
+
+  const project = useProject();
+  let boundedSize: number | undefined = undefined;
+  if (size) {
+    boundedSize = Math.max(Math.min(MAX_SIZE, size), MIN_SIZE) - 24;
+    if (boundedSize < MIN_SIZE) {
+      boundedSize = undefined;
+    }
+  }
+  const { updateScreenshots } = useTranslationsActions();
+
+  const oneOnly = screenshots.length === 1 && boundedSize && oneBig;
+  const [detailData, setDetailData] = useState<number>();
+
+  const deleteLoadable = useApiMutation({
+    url: '/v2/projects/{projectId}/keys/{keyId}/screenshots/{ids}',
+    method: 'delete',
+  });
+
+  const handleDelete = (id: number) => {
+    deleteLoadable.mutate(
+      {
+        path: { projectId: project.id, ids: [id], keyId },
+      },
+      {
+        onSuccess() {
+          updateScreenshots({
+            keyId,
+            screenshots(data) {
+              return data.filter((sc) => sc.id !== id);
+            },
+          });
+        },
+      }
+    );
+  };
+
+  const calculatedHeight = Math.min(
+    Math.max(...screenshots.map((sc) => 100 / (sc.width! / sc.height!))),
+    100
+  );
+
+  const screenshotsMapped = screenshots.map((sc) => {
+    return {
+      id: sc.id,
+      src: sc.fileUrl,
+      width: sc.width,
+      height: sc.height,
+      highlightedKeyId: keyId,
+      keyReferences: sc.keyReferences,
+    };
+  });
+
+  const [scrollLeft, scrollRight] = useScrollStatus(containerRef, [
+    width,
+    screenshots,
+  ]);
+
+  function handleScrollLeft() {
+    const element = containerRef.current;
+    if (element) {
+      const scrollLeft = element?.scrollLeft;
+      element?.scroll({ left: scrollLeft - 100, behavior: 'smooth' });
+    }
+  }
+
+  function handleScrollRight() {
+    const element = containerRef.current;
+    if (element) {
+      const scrollLeft = element?.scrollLeft;
+      element?.scroll({ left: scrollLeft + 100, behavior: 'smooth' });
+    }
+  }
+
+  return (
+    <StyledContainer
+      {...{ sx }}
+      className={clsx({ scrollLeft, scrollRight })}
+      onClick={stopAndPrevent()}
+    >
+      <StyledScrollWrapper ref={containerRef}>
+        {screenshots.map((sc, index) => {
```

Comment:
Can we move this to separate component so the file is not so big and we don't have to do so much logic in the JSX?